From: Thorsten Glaser <tg@debian.org>
Date: Tue, 27 Mar 2012 20:14:22 +0000
Subject: make select testcase actually useful

What's the use of a testcase run at build if it always returns 0,
even in the face of an error?

Index: pkg-dietlibc/test/select.c
===================================================================
--- pkg-dietlibc.orig/test/select.c	2012-03-25 15:07:46.000000000 +0000
+++ pkg-dietlibc/test/select.c	2012-03-27 20:08:53.000000000 +0000
@@ -8,6 +8,9 @@
   struct timeval tv;
   FD_ZERO(&f);
   tv.tv_sec=3; tv.tv_usec=0;
-  select(1,&f,0,0,&tv);
+  if (select(1,&f,0,0,&tv) == -1) {
+    perror("select");
+    return 1;
+  }
   return 0;
 }
